landing-zone/dynamic_values/f5_vsi.tf

```diff
@@ -0,0 +1,130 @@
+  do_declaration_url      = lookup(var.f5_template_data, "do_declaration_url", null) == null ? "null" : lookup(var.f5_template_data, "do_declaration_url", null)
+  as3_declaration_url     = lookup(var.f5_template_data, "as3_declaration_url", null) == null ? "null" : lookup(var.f5_template_data, "as3_declaration_url", null)
+  ts_declaration_url      = lookup(var.f5_template_data, "ts_declaration_url", null) == null ? "null" : lookup(var.f5_template_data, "ts_declaration_url", null)
+  phone_home_url          = lookup(var.f5_template_data, "phone_home_url", null) == null ? "null" : lookup(var.f5_template_data, "phone_home_url", null)
+  template_source         = lookup(var.f5_template_data, "template_source", null) == null ? "null" : lookup(var.f5_template_data, "template_source", null)
+  template_version        = lookup(var.f5_template_data, "template_version", null) == null ? "null" : lookup(var.f5_template_data, "template_version", null)
+  app_id                  = lookup(var.f5_template_data, "app_id", null) == null ? "null" : lookup(var.f5_template_data, "app_id", null)
+  tgactive_url            = lookup(var.f5_template_data, "tgactive_url", null) == null ? "null" : lookup(var.f5_template_data, "tgactive_url", null)
+  tgstandby_url           = lookup(var.f5_template_data, "tgstandby_url", null) == null ? "null" : lookup(var.f5_template_data, "tgstandby_url", null)
+  tgrefresh_url           = lookup(var.f5_template_data, "tgrefresh_url", null) == null ? "null" : lookup(var.f5_template_data, "tgrefresh_url", null)
+}
+
+
+##############################################################################
+
+##############################################################################
+# [Unit Test] F5 and Template
+##############################################################################
+
+module "ut_f5" {
+  source = "./config_modules/f5"
+  prefix = "ut"
+  region = "us-south"
+  f5_vsi = [
+    {
+      name                = "f5-zone-1"
+      primary_subnet_name = "subnet-1"
+      vpc_name            = "test"
+      secondary_subnet_names = [
+        "subnet-f5-1",
+        "subnet-f5-2"
+      ]
+      hostname = "f5-ve-01"
+      domain   = "local"
+    }
+  ]
+  vpc_modules = {
+    test = {
+      vpc_id = "1234"
+      subnet_zone_list = [
+        {
+          name = "ut-test-subnet-1"
+          id   = "1-id"
+          zone = "1-zone"
+          cidr = "10.10.10.10/10"
+        },
+        {
+          name = "ut-test-subnet-f5-1"
+          id   = "1-id"
+          zone = "1-zone"
+          cidr = "10.10.10.10/10"
+        },
+        {
+          name = "ut-test-subnet-f5-2"
+          id   = "1-id"
+          zone = "1-zone"
+          cidr = "10.10.10.10/10"
+        }
+      ]
+    }
+  }
+  f5_template_data = {
+    tmos_admin_password     = "Gooooooooooooooooodpass1"
```

Comment:
For discussion, do we want to hardcode a password value here? I expect not...

Reply:
Agree - we should force users to pass a password here by not setting default ones, so that customers do not end up with a public password they have no knowledge of